frontend/javascripts/libs/toast.tsx

```diff
@@ -112,13 +112,29 @@ const Toast = {
     }
 
     notification[type](toastConfig);
+
+    // Make sure that toasts don't just dissapear while user has WK in background tab (e.g. while uploading large dataset).
+    // Most browsers pause requestAnimationFrame() if the current tab is not active, but Firefox does not seem to do that.
+    if (!sticky && key != null) {
+      requestAnimationFrame(async () => {
+        // ensure tab is active
+        const timeoutAfterTabSwitch = timeout >= 1000 ? 1000 : 0;
+        await sleep(timeout - timeoutAfterTabSwitch);
```

Comment:
how about:

```suggestion
        const timeoutAfterTabSwitch = timeout / 2; // todo: maybe another var name would make more sense now
        await sleep(timeoutAfterTabSwitch);
```

this effectively splits the sleep into two equally long phases which can only start when the animation frame is fulfilled each. I find that conceptionally more intuitive.